### ISSUE TITLE
Cant detect pretrained classes 

### ISSUE THREAD

**Issue by Tzoulio** [ultralytics/yolov5#4578]
## ❔After training on the yolov5s weights i can detect my custom classes fine but i am unable to detect the pretrained ones. 

I created a data.yaml file in which i included my custom classes plus i left some classes like person or dinning table which the pretrained weights already knew. The dataset im using has no people or dinning table images in it I thought i didnt need to have any maybe i was wrong? Aren't the yolov5s weights trained on the coco dataset?

**Comment by glenn-jocher** [comment]
@Tzoulio your dataset should include data for **all classes** you expect to detect during deployment.

**Comment by Tzoulio** [comment]
> @Tzoulio your dataset should include data for **all classes** you expect to detect during deployment.

Don't the pretrained weights already have that "knoweledge"? I thought that was the point of using them besides training time.

**Comment by Tzoulio** [comment]
If I make a prediction using the yolov5s weights they can detect a person fine. However, when using the weights that come from trainining yolov5s I cannot detect the same picture of the person. Is it a case of the pretrained weights updating maybe? Should I freeze them so they stay the same? I am only guessing trying to provide as much info as possible thanks in advance..

**Comment by glenn-jocher** [comment]
@Tzoulio your assumptions about how ML works is not correct. If you train on new classes your old classes are not retained. 

You either have your cake or eat it, but not both.

**Comment by Tzoulio** [comment]
So if I train a model using pretrained weights that know an "x" number of classes and I want to add one more class to that list, my dataset should have images of all the previous classes plus the new one?

**Comment by glenn-jocher** [comment]
@Tzoulio yes, your dataset should include data for all classes you expect to detect during deployment.

You can train on multiple datasets simultaneously, though you need to ensure your classes do not intersect among your datasets. If you want to train COCO + custom, then start labelling your custom data from class 80 onward, and then group them in your data.yaml. See GlobalWheat yaml for an idea:

https://github.com/ultralytics/yolov5/blob/d7aa3f153d049b89267b1b594a481a1a27fe27e6/data/GlobalWheat2020.yaml#L10-L27

**Comment by Tzoulio** [comment]
Thank you for the clarification. One last question, if I want to train on coco+custom besides adding the data labels from class 80 onward, do I have to use the coco dataset together with mine?

**Comment by glenn-jocher** [comment]
@Tzoulio yes

**Comment by Tzoulio** [comment]
Thank you for replying and helping me out. Do you have any reading material to fix my incorrect ML knowledge?

**Comment by glenn-jocher** [comment]
@Tzoulio hey buddy, I would just start from the Train Custom Data tutorial, and then once you get good results there with COCO128, move on to your custom dataset by itself (see **Tips for Best Training Results**). Once you've got good results there you might wander to more advanced topics like mixing datasets etc.

## YOLOv5 Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://docs.ultralytics.com/yolov5/tutorials/model_export8)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [TorchScript, ONNX, CoreML Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) 🚀
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)
